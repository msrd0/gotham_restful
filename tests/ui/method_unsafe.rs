#[macro_use] extern crate gotham_restful;

#[derive(Resource)]
#[resource(read_all)]
struct FooResource;

#[read_all(FooResource)]
unsafe fn read_all()
{
}

fn main()
{
}
