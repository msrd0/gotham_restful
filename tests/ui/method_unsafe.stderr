error: Resource methods must not be unsafe
 --> $DIR/method_unsafe.rs:8:1
  |
8 | unsafe fn read_all()
  | ^^^^^^

error[E0433]: failed to resolve: use of undeclared type or module `_gotham_restful_resource_foo_resource_method_read_all`
 --> $DIR/method_unsafe.rs:3:10
  |
3 | #[derive(Resource)]
  |          ^^^^^^^^ use of undeclared type or module `_gotham_restful_resource_foo_resource_method_read_all`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
